security/advancedtls/advancedtls.go

```diff
@@ -0,0 +1,377 @@
+/*
+ *
+ * Copyright 2019 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+// Package advancedtls is a utility library containing functions to construct
+// credentials.TransportCredentials that can perform credential reloading and custom
+// server authorization.
+// All APIs in this package are EXPERIMENTAL.
+package advancedtls
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"net"
+	"syscall"
+	"time"
+
+	"google.golang.org/grpc/credentials"
+)
+
+type sysConn = syscall.Conn
+
+// CustomVerificationFunc is the function defined by users to perform custom server
+// authorization.
+type CustomVerificationFunc func(serverName string,
+	rawCerts [][]byte, verifiedChains [][]*x509.Certificate) error
+
+// RootCertificateOptions contains a field and a function for obtaining root trust certificates.
+// It is used by both ClientOptions and ServerOptions.
+type RootCertificateOptions struct {
+	// ---------------General Rules For Root Certificate Setting------------------------------------
+	// RootCACerts or GetRootCAs indicates the the certificates trusted by the client side. The rules
+	// for setting these two fields are:
+	// Either RootCACerts or GetRootCAs must be set; the other will be ignored
+	//----------------------------------------------------------------------------------------------
```

Comment:
This comment looks like it belongs with the struct and not the RootCACerts field.